packages/@orbit/jsonapi/src/jsonapi-serializer.ts

```diff
@@ -309,6 +325,104 @@ export class JSONAPISerializer
     return result;
   }
 
+  deserializeOperationsDocument(
+    document: ResourceOperationsDocument
+  ): RecordOperation[] {
+    return this.deserializeOperations(document.operations);
+  }
+
+  deserializeOperations(operations: ResourceOperation[]): RecordOperation[] {
+    return operations.map(operation => {
+      if (isAddOperation(operation)) {
+        return this.deserializeAddOperation(operation);
+      } else if (isUpdateOperation(operation)) {
+        return this.deserializeUpdateOperation(operation);
+      } else if (isRemoveOperation(operation)) {
+        return this.deserializeRemoveOperation(operation);
+      } else {
+        throw new Error(
+          `JSONAPISerializer: "get" operation recieved but only "add", "update" and "remove" operations are supported as input at this time.`
+        );
+      }
+    });
+  }
+
+  deserializeAddOperation(
+    operation: AddResourceOperation | AddToRelatedResourcesOperation
+  ): AddToRelatedRecordsOperation {
```

Comment:
Shouldn't this method also handle `AddResourceOperation` ? Probably just needs an `else` clause, right?

Reply:
Yes. I actually have a question here. An `add` operation can have or not an `id`. Should we handle add without `id`? What should we do in this case? Call `schema.generateId()`?

Reply:
When _receiving and deserializing_ a document that contains operations, then I think we can guarantee that resources in operations will always have an `id` (since a server will be emitting these operations). If no `id` is present, I would expect an error to be raised.

However, we can make no such guarantees or expectations with serialization.